### PR TITLE
chore(ci): Run CLI tests on Mac and Windows

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes https://github.com/cloudquery/cloudquery/issues/1803

### CODE REVIEW

File: .github/workflows/wait_for_required_workflows.yml [hunk 1]
```diff
@@ -61,7 +61,7 @@ jobs:
               })
               const runs = checkRuns.map(({ name, conclusion }) => ({ name, conclusion }))
               console.log(`Got the following check runs: ${JSON.stringify(runs)}`)
-              const matchingRuns = runs.filter(({ name }) => actions.includes(name))
```
Comment (erezrokah) [inline .github/workflows/wait_for_required_workflows.yml:64]: With the matrix, names are `cli (<os>)`, e.g. `cli (macos-latest)`

File: .github/workflows/wait_for_required_workflows.yml [hunk 2]
```diff
@@ -61,7 +61,7 @@ jobs:
...
+              const matchingRuns = runs.filter(({ name }) => actions.startsWith(name))
```
Comment (erezrokah) [inline .github/workflows/wait_for_required_workflows.yml:64]: With the matrix, names are `cli (<os>)`, e.g. `cli (macos-latest)`

**Comment by erezrokah** [comment]
Tests are failing due to https://github.com/cloudquery/cloudquery/issues/2206

**Review by yevgenypats — APPROVED** [review]

**Comment by erezrokah** [comment]
Still fails on Windows with a different error 😄

**Comment by erezrokah** [comment]
For some reason testing on Windows hangs now, so can't be sure it's working.

**Pull request merged**